index.js

```diff
@@ -63,11 +51,115 @@ async function gatherResponse(response) {
   }
 }
 
+// Validate the entries in the form
+function validateFormData(bodydata)
+{
+  if (!bodydata) return false
+  if (!bodydata.name) return false
+
+  // Taken from http://emailregex.com/
+  const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+  if (!bodydata.address || !emailRegex.test(bodydata.address)) return false
+
+  return true
+}
+
 async function handleRequest(request) {
-  const formData = await request.formData()
-  const bodydata = {}
-  for (const entry of formData.entries()) {
-    bodydata[entry[0]] = entry[1]
+
+  try
+  {
+    //
+    // \TODO !! Uncomment the origin checking for the live version !!
+    //
+    // Validate the origin of the request
+    if( request.method === 'POST' ) // && request.url.hostname === '...' && request.url.pathname === '/add')
+    {
+      const formData = await request.formData()
+      const bodydata = {}
+      const listsToSubscribeTo = [] // Empty array to hold all the email lists to sign up to
+      for (const entry of formData.entries()) {
+        bodydata[entry[0]] = entry[1]
```

Comment:
This allows to sign up for any mailing list on the server if its address is known. Instead we should explicitly match the entries against the lists we have.

Reply:
Thanks for catching that! I was trying to avoid having to hard-code which lists can be signed up for this way. How would you suggest patching this?

Reply:
Hard-code for now, eventually rebuilt the js file in actions during deploying.